### ISSUE TITLE
resilience4j-bom is missing from Maven central

### ISSUE THREAD

**Issue by BenEfrati** [resilience4j/resilience4j#310]
Hi,
The resilience4j-bom not exists in maven central https://search.maven.org/
From release notes:

> PR #277: Generate BOM for resilience4j

Thanks,
Ben

**Comment by storozhukBM** [comment]
Hi @BenEfrati,
Yes, you're right. We've noticed that also. It seems to be an issue with our Binrtay publishing plugin, and it is not reproducible during snapshot publishing, so it's pretty hard to debug. During next release I'll try to update plugin version and use some debug logs to troubleshoot that. If you have some other suggestions please let me know.

**Comment by drtrang** [comment]
The resilience4j-bom still not exist in latest version(0.14.1)

**Comment by RobWin** [comment]
Yes, we are upgrading to Gradle 5 at the moment. I will have a look at it again and try to use new Gradle features.
https://github.com/gradle/gradle/blob/master/subprojects/dependency-management/preview-features.adoc#maven-bom-files-and-dependencymanagement

**Comment by leonard84** [comment]
@RobWin why not publish it manually, for the missing versions?

**Comment by mbazos** [comment]
Yeah for the time being can you manually upload the latest ressilence4j to maven repo as 1.0.0 seems to be missing:

https://mvnrepository.com/artifact/io.github.resilience4j/

**Comment by RobWin** [comment]
I tried but it failed. I have to look at the gradle build again.

**Comment by mbazos** [comment]
@RobWin thanks for looking into this! If I have some spare time I don't mind to help, can you share the error you are running into with gradle?

**Comment by flozano** [comment]
Still missing

**Comment by davidkarlsen** [comment]
@RobWin maybe you can pick up some info from https://github.com/spring-projects/spring-boot/blob/master/spring-boot-project/spring-boot-dependencies/build.gradle ? I think this would be a handy feature to have (a bom)

**Comment by mbazos** [comment]
@davidkarlsen & @RobWin I have built some boms before (internal for my company), if you don't mind I could probably spend a little time on this and submit a PR.

**Comment by RobWin** [comment]
Any help is welcome ;)

**Comment by mbazos** [comment]
@RobWin I see there actually is a https://github.com/resilience4j/resilience4j/tree/master/resilience4j-bom module, does this not do what @davidkarlsen was asking for?

**Comment by davidkarlsen** [comment]
Yes, but it's not @ central: https://repo1.maven.org/maven2/io/github/resilience4j/

**Comment by Romeh** [comment]
@mbazos can u please review https://github.com/resilience4j/resilience4j/tree/master/resilience4j-bom  module and verify the gradle config there so we can see why it is not published to maven central properly ?

**Comment by mbazos** [comment]
I will take a look

**Comment by mbazos** [comment]
I think the problem might be here in the root build.gradle

```
ext {
    coreProjects = subprojects.findAll {
        p -> !p.name.contains("documentation") && !p.name.endsWith("-bom")
    }
}
```

or

```
subprojects {
    jacocoTestReport {
        reports {
            xml.enabled true
        }
    }

    afterEvaluate {
        // exclude subprojects that don't produce a jar file or by design.
        if (!project.name.equals('resilience4j-bom') && !project.name.equals('resilience4j-documentation')
                && !project.name.equals('resilience4j-test')) {
            jar {
                inputs.property('moduleName', moduleName)
                manifest.attributes(
                        'Automatic-Module-Name': moduleName
                )
            }
        }
    }
}
```
I was able to build the project and publish to my local m2 and see the file correctly `resilience4j-bom-1.4.0-SNAPSHOT.pom` I am not sure how to simulate actually pushing to maven central besides pushing to my local m2 repo (which looks correct)

**Comment by Romeh** [comment]
@mbazos  i will give it a try based into ur remarks tomorrow thx a lot for your feedback man !

**Comment by fschmager** [comment]
"tomorrow" was a month ago. Any news on this one? 

Love resilience4j and the bom would be the cherry on top.

**Comment by RobWin** [comment]
I finally had time to fix the build script some days ago. -> https://github.com/resilience4j/resilience4j/pull/965

I added the platform gradle plugin to publish a platform BOM. Unfortunately I did it after releasing 1.4.0.
The next release should finally have a BOM published automatically.

**Comment by mbazos** [comment]
Thanks @RobWin

**Comment by RobWin** [comment]
Finally it works. The bom has been automatically released with v1.5.0 ;)

```
dependencies {
	implementation platform('io.github.resilience4j:resilience4j-bom:1.5.0')
	implementation("io.github.resilience4j:resilience4j-spring-boot2")
	implementation("io.github.resilience4j:resilience4j-reactor")
	implementation("io.micrometer:micrometer-registry-prometheus")
}
```
